### ISSUE TITLE
How to increase timeout of failed jobs

### ISSUE THREAD

**Issue by eriktelepovsky** [rq/django-rq#455]
Hello. Is there a way how to change/increase timeout of failed jobs?

My jobs fail with exception:

```
rq.timeouts.JobTimeoutException: Task exceeded maximum timeout value
```

so I increased the ``DEFAULT_TIMEOUT`` in ``RQ_QUEUES`` setting, but every time I requeue failed jobs, the new setting is ignored and and is used previous timeout value.

**Comment by selwin** [comment]
Are you getting the queue using `django_rq.get_queue()`?

**Comment by eriktelepovsky** [comment]
Hi @selwin. Do you mean when I requeue the failed jobs? I don't do it programmatically. I use django admin action:

![image](https://user-images.githubusercontent.com/2685808/99670519-f597c480-2a70-11eb-88f1-568a7536225e.png)

**Comment by selwin** [comment]
No, I meant when you're enqueueing the job originally.

**Comment by eriktelepovsky** [comment]
I'm using method decorator:

```
from django_rq import job

@job('long_running')
def my_method()
   ...
```

so yes, I presume it is using ``get_queue()``

**Comment by selwin** [comment]
Yeah, so I think that should work. Are your settings.py configured correctly? Alternatively you can also do `@job('long_running', timeout=1000)`

**Comment by eriktelepovsky** [comment]
When I increase the timeout in the settings, it is applied to all new delayed jobs successfully. But it is not applied to the requeued failed jobs. Old timeout value is used.

**Comment by selwin** [comment]
Ah ok. Then what’s needed is a feature that allows you to update job.timeout prior to requeueing it.

Unfortunately this feature is not in DjangoRQ yet. If you want, I’ll be happy to accept a PR for this.

**Comment by eriktelepovsky** [comment]
I can look into it. Can you guide me where to start?

**Comment by selwin** [comment]
Create an `edit_job_timeout` view in https://github.com/rq/django-rq/blob/master/django_rq/views.py#L302

This view should be reachable via job detail page:
![Selection_281](https://user-images.githubusercontent.com/188658/100594844-c4e74300-332c-11eb-9d72-4ba447667199.jpg)

**Comment by eriktelepovsky** [comment]
And what about automatic timeout update according to the current settings when job is requeued?

**Comment by selwin** [comment]
It won't be automated, since the `job` object is already created and stored in Redis, complete with it's timeout. Automatically updating a value that's already stored, whether it's in DB or Redis whenever a configuration is changed is not recommended.

**Comment by eriktelepovsky** [comment]
I created a PR #458
